xmodule/capa_block.py

```diff
@@ -2160,6 +2247,7 @@ def rescore(self, only_if_higher=False):
         # rescoring should have no effect on attempts, so don't
         # need to increment here, or mark done.  Just save.
         self.set_state_from_lcp()
+        self.set_score(calculated_score)
```

Comment:
should this happen even with the flag off?

Reply:
Yes, this allows the update of the current score with the new score calculated.

Reply:
After doing some tests I finally removed this code, since the score is updated when the `publish_grade` method is called, so this change is not necessary.